backend/question_service/middleware/isAdminCheck.js

```diff
@@ -0,0 +1,37 @@
+const isAdminCheck = async (req, res, next) => {
+    // Extract the token from the request headers
+    const token = req.headers['authorization']?.split(' ')[1];
+    
+    if (!token) {
+        return res.status(401).json({ error: "Authorization token is required." });
+    }
+
+    try {
+        // Make a request to the user-service to check admin status
+        const userServiceBackendUrl = process.env.USER_SERVICE_BACKEND_URL || "http://localhost:5001";
```

Comment:
The current value defined for `USER_SERVICE_BACKEND_URL` is `http://user_service:5001/verify-token` in the `.env` file, which is not the correct URL to be used here.

One suggestion is, refactor that URL's value into a generic `http://user_service:5001` in the `.env` file. This way, the path behind like `verify-token` or `checkAdminStatus` can then be added individually at each relevant file.

Then need to find all affected files that used that URL and update its fetch method to manually add in the behind path.

Also will need to update at 4 other files (for every file that is affected by the change).

1) .env file 
2) Dockerfile
3) docker compose.yml
4) README.md

Reply:
Thanks so much for pointing this out. I have made the necessary edits.  Just to be clear, the `docker-compose.yml` and `Dockerfile` only has `USER_SERVICE_BACKEND_URL` as a referenced variable, so should be fine without changes?